woocommerce/woocommerce-ios
# [Woo POS] Avoid initial view model unwanted recreation

This PR addresses the issue discussed in slack, here: p1715934814114829-slack-C025A8VV728 . #12761 should be merged first

## Description
The initial view model that drives the pos dashboard is currently created inside the `PointOfSaleEntryPointView`, which is recreated multiple times, triggering a view model recreation with it.

This PR addresses this problem by extracting the view model initialization a few levels up the chain, to the `HubMenuViewController`, to assure it's only created once through the app's lifecycle, and then we inject it into the proper view.

## Question
I'm not quite sure if we should compose the view model and its dependencies on a different manner. Doing so in the `HubMenuViewController` doesn't feel quite right, but does follow the same pattern we already use for creating the `HubMenuViewModel`. 

Perhaps moving its composition to the `HubMenuCoordinator` or the `MainTabBarController` would make sense? Despite POS not being a "tab", it is intended to be a different mode.

## Testing instructions
- In the POS view, tap on the `Filter` button and observe that the modal is not automatically dismissed anymore.
- Alternatively, add a `debugprint` on view model init and observe that is only created once.
